mesonbuild/compilers/mixins/clang.py

```diff
@@ -139,6 +140,8 @@ def gen_vs_module_defs_args(self, defsfile: str) -> T.List[str]:
             # With MSVC, DLLs only export symbols that are explicitly exported,
             # so if a module defs file is specified, we use that to export symbols
             return ['-Wl,/DEF:' + defsfile]
+        if isinstance(self.linker, ClangClDynamicLinker):
+            return ['-Wl,/DEF:' + re.sub(R'(?<!\\)\\(?!\\)', R'\\\\', defsfile.replace('/', '\\'))]
```

Comment:
I don't love the use of the regex here, they're slow and difficult to understand. would just:
```python
if os.altsep:
   defsfile = defsfile.replace(os.altsep, os.sep)
return [... + defsfile.ecnode('unicode-escape')]
```
work here? (might have os.altsep and os.sep backwards.